src/node_task_runner.cc

```diff
@@ -240,19 +219,66 @@ void ProcessRunner::Run() {
   uv_run(loop_, UV_RUN_DEFAULT);
 }
 
+std::optional<std::tuple<std::filesystem::path, std::string, std::string>>
+FindPackageJson(const std::filesystem::path& cwd) {
+  // Two different paths are created only for improving readability.
+  // Otherwise, it is not needed.
+  auto directory_path = cwd;
+  auto package_json_path = directory_path / "package.json";
+  std::string raw_content;
+  std::string path_env_var =
+      (directory_path / "node_modules" / ".bin").string() + env_var_separator;
+
+  USE(ReadFileSync(&raw_content, package_json_path.c_str()));
+
+  do {
+    directory_path = directory_path.parent_path();
+
+    // Always append "node_modules/.bin" to the env var.
+    // This is in par with existing package managers.
+    path_env_var +=
+        (directory_path / "node_modules" / ".bin").string() + env_var_separator;
```

Comment:
it's probably better to use the correct path separator on windows

Reply:
/ is an overloaded operator. string() converts it into native platform specific path in std::filesystem